txsni/snimap.py

```diff
@@ -93,8 +115,13 @@ def set_npn_select_callback(self, cb):
         return self._obj.set_npn_select_callback(cb)
 
     def set_alpn_select_callback(self, cb):
-        self._factory._alpnSelectCallbackForContext(self._obj, cb)
-        return self._obj.set_alpn_select_callback(cb)
+
+        def alpn_callback(connection, protocols):
+            """wrapped alpn_select_callback"""
+            return self._factory.selectAlpn(lambda: cb(connection, protocols), connection, protocols)
```

Comment:
Is this actually helpful given the problems within OpenSSL?

Reply:
Yes. The wrapped function says "yes, I will speak acme-tls/1" when the client has asked for it. We can do that without including acme-tls/1 in the list used by the default callback.